w-11599419-untilSuccessful-duke (#2430)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/until-successful-scope.adoc

endif::[]
:keywords: anypoint studio, studio, mule, until successful, reattempts, retry

The Until Successful scope executes processors within it, in order, until they

> all succeed or exhaust the maximum number of retries. Like all Core components

Author: ```suggestion
all succeed or the scope exhausts the maximum number of retries. Like all Core components
```

endif::[]
:keywords: anypoint studio, studio, mule, until successful, reattempts, retry

The Until Successful scope executes processors within it, in order, until they

> all succeed or the scope exhausts the maximum number of retries. Like all Core components

Author: ```suggestion
all succeed or the scope exhausts the maximum number of retries. 
```

> other than Async, Until Successful runs synchronously. If any processor

Author: ```suggestion
Until Successful runs synchronously. If any processor
```

within the scope fails to connect or to produce a successful result,
Until Successful retries all the processors within it, including the one that failed, until all configured retries
are exhausted. If a retry succeeds, the scope proceeds to the next
component. If the final retry does not succeed, Until Successful produces an error.

> The scope routes a message to its processors in an asynchronous manner. Routing is successful if no exception is raised 

Author: ```suggestion
Routing is successful if no exception is raised 
```

> The scope to route a message to its message processors in an asynchronous manner. Routing is successful if no exception is raised 

Author: ```suggestion
The scope routes a message to its processors in an asynchronous manner. Routing is successful if no exception is raised 
```

You can configure the following attributes in the Until Successful scope:

[%header,cols="1,1,3"]
|===

> |Field | XML | Description

Author: This was just reorg'd to add a column, not rewritten.

* If the FTP Write operation fails
+
The Until Successful scope retries the operation every `3000` milliseconds until the operation succeeds, with a limit of `5` retries. If the last execution fails, the scope throws a `MULE:RETRY_EXHAUSTED` error. Then, the `<on-error-continue>` handles the error and executes the Logger with the message: `File upload failed`. 
+

> Before throwing `MULE:RETRY_EXHAUSTED`, Mule also logs each unsuccessful attempt to retry execution of the event, for example:

Author: ```suggestion
Before throwing `MULE:RETRY_EXHAUSTED`, Mule also logs each unsuccessful attempt to retry execution of the processors on the Mule event, for example:
```
